### PR TITLE
itkdev Feature/5371 blocked patron

### PR DESCRIPTION
**Pull request by sinejespersen**
#### Link to issue

https://platform.dandigbib.org/issues/5371

#### Description

add strings and config for recently merged (in dpl-react) blocked patron

#### Screenshot of the result

image

### CODE REVIEW

**Review by cableman — APPROVED** [review]
Looks fine, just need to revert the react URL in composer.json before external review.

**Comment by sinejespersen** [comment]
> Looks fine, just need to revert the react URL in composer.json before external review.

ofc

**Review by spaceo — CHANGES_REQUESTED** [review]
A dependency needs to be changed
and I have a question...

File: web/modules/custom/dpl_library_agency/src/Form/GeneralSettingsForm.php [hunk 1]
```diff
@@ -144,6 +144,27 @@ public function buildForm(array $form, FormStateInterface $form_state): array {
...
+      '#description' => $this->t('The link to redirect the blocked user to'),
+      '#default_value' => $config->get('redirect_on_blocked_url') ?? '',
+    ];
+
+    $form['settings']['blocked_user']['blocked_patron_e_link_url'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Blocked user link for modal'),
+      '#description' => $this->t('If a user has blocked status e, this link appears in the modal'),
+      '#default_value' => $config->get('blocked_patron_e_link_url') ?? '',
+    ];
```
Comment (spaceo) [inline web/modules/custom/dpl_library_agency/src/Form/GeneralSettingsForm.php:166]: I think you should use the Form API type: 'url' here:
https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Render%21Element%21Url.php/class/Url/

Comment (spaceo) [reply web/modules/custom/dpl_library_agency/src/Form/GeneralSettingsForm.php:166]: Then you get the url validation for free! :)

File: composer.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
                 "version": "0.3.0-rc28",
                 "type": "drupal-library",
                 "dist": {
-                    "url": "https://github.com/danskernesdigitalebibliotek/dpl-react/releases/download/release-develop/dist.zip",
+                    "url": "https://github.com/itk-dev/dpl-react/releases/download/release-feature%2Fblocked-null-value/dist.zip",
```
[the remark this answers is not in the document]
Comment (sinejespersen) [reply composer.json:23]: Yes!

File: web/modules/custom/dpl_library_agency/src/Form/GeneralSettingsForm.php (hunk 1 above)
Comment (sinejespersen) [reply web/modules/custom/dpl_library_agency/src/Form/GeneralSettingsForm.php]: Changed

File: web/modules/custom/dpl_loans/src/Plugin/Block/LoanListBlock.php [hunk 3]
```diff
@@ -75,7 +75,7 @@ public static function create(ContainerInterface $container, array $configuratio
    */
   public function getThresholdConfig(): string {
     $generalSettings = $this->configFactory->get('dpl_library_agency.general_settings');
-    return $generalSettings->get('threshold_config');
+    return $generalSettings->get('threshold_config') ?? "{ 'colorThresholds': { 'danger': '0', 'warning': '6' } }";
```
Comment (spaceo) [inline web/modules/custom/dpl_loans/src/Plugin/Block/LoanListBlock.php:78]: I am not a fan of hardcodet config if it is not centralized.
It should be possible to store as exported config though.

**Review by spaceo — CHANGES_REQUESTED** [review]
Just spotted a minor config matter...

File: web/modules/custom/dpl_loans/src/Plugin/Block/LoanListBlock.php (hunk 3 above)
Comment (sinejespersen) [reply web/modules/custom/dpl_loans/src/Plugin/Block/LoanListBlock.php:78]: https://github.com/danskernesdigitalebibliotek/dpl-cms/pull/303/commits/dd82a896b9160af77207107dd55da9ec69c14dbb

Comment (sinejespersen) [reply web/modules/custom/dpl_loans/src/Plugin/Block/LoanListBlock.php:78]: wie so?

**Pull request merged**